sql/core/src/main/scala/org/apache/spark/sql/columnar/InMemoryColumnarTableScan.scala

```diff
@@ -268,16 +268,21 @@ private[sql] case class InMemoryColumnarTableScan(
       readBatches.setValue(0)
     }
 
-    relation.cachedColumnBuffers.mapPartitions { cachedBatchIterator =>
-      val partitionFilter = newPredicate(
-        partitionFilters.reduceOption(And).getOrElse(Literal(true)),
-        relation.partitionStatistics.schema)
+    val schema = relation.partitionStatistics.schema
+    val schemaIndex = schema.zipWithIndex
+    val relOutput = relation.output
+    val buffers = relation.cachedColumnBuffers
```

Comment:
Could you please add a comment above this hunk to indicate that these variables are added to avoid unnecessary serialization?